LDEventSource/LDEventParser.m

```diff
@@ -66,14 +79,57 @@ -(LDEvent*)eventFromString:(NSString*)eventString {
                 } else if ([key isEqualToString:LDEventIDKey]) {
                     event.id = value;
                 } else if ([key isEqualToString:LDEventRetryKey]) {
-                    self.foundRetryInterval = [value doubleValue];
+                    if ([value isKindOfClass:[NSNumber class]]) {
+                        self.retryInterval = @([value doubleValue]);
+                    }
                 }
             }
         }
     }
-    if (self.remainingEventString.length == 0) {
-        self.remainingEventString = nil;
-    }
-    return event;
+}
+
+//extracts lines from the first thru the first empty line
+-(NSArray<NSString*>*)linesToParseFromEventString {
```

Comment:
#picky could we add nullability annotations to this so it's more clear that `nil` is a valid return value?